parser/parser_test.go

```diff
@@ -301,6 +315,189 @@ func TestMemberExpr(t *testing.T) {
 	})
 }
 
+func TestVarDeclarationErrors(t *testing.T) {
+	runTests(t, []TestCase{
+		{
+			name: "NotIdentifierAfterVarDecl",
+			code: "var 1234;",
+			fail: true,
+		},
+		{
+			name: "VarWithoutIdentifier",
+			code: "var",
+			fail: true,
+		},
+		{
+			name: "EOFAfterIdentifier",
+			code: "var x",
+			fail: true,
+		},
+		{
+			name: "EOFAfterInitializer",
+			code: "var x =",
+			fail: true,
+		},
+		{
+			name: "InvalidAssignExpression",
+			code: "var a = var;",
+			fail: true,
+		},
+		{
+			name: "InvalidAssignInitializer",
+			code: "var a ! 5;",
+			fail: true,
+		},
+		{
+			name: "InvalidVarOnMultipleInits",
+			code: "var a = 5, var x = 6;",
+			fail: true,
+		},
+		{
+			name: "InvalidAssignOnMultipleInits",
+			code: "var a = 5, x ! 3",
+			fail: true,
+		},
+		{
+			name: "InvalidSeparatorForMultipleInits",
+			code: "var d = 6 : x = 1",
+			fail: true,
+		},
+		{
+			name: "InvalidFuncall",
+			code: "var d = lala(666",
+			fail: true,
+		},
+		{
+			name: "InvalidMemberAccess",
```

Comment:
this test shoulnt be in together with memberexpr tests?

Reply:
Not sure, they are just as about member expression as they are about variable initialization. Actually the objective of these tests is to check that if the parser of the member access (the one that already has been tested isolated) fails the parser for variable statements checks that error and handles it properly, so it seemed like here would be the best place.

Reply:
Do you are saying that this just test if errors are being checked ?